cs3api4lab/config/config_manager.py

```diff
@@ -9,34 +9,23 @@ class Config(LoggingConfigurable):
     config = None
     __config_dir = "\\jupyter-config"
     __config_file_name = 'jupyter_cs3_config'
-    __env_names = {"reva_host",
-                   "auth_token_validity",
-                   "endpoint",
-                   "home_dir",
-                   "chunk_size",
-                   "secure_channel",
-                   "client_cert",
-                   "client_key",
-                   "ca_cert",
-                   "client_id",
-                   "client_secret",
-                   "login_type"}
+    __env_vars_req = {"reva_host",
+                      "client_id",
+                      "client_secret"}
+    __env_vars_opt = {"auth_token_validity",
+                      "endpoint",
+                      "home_dir",
+                      "chunk_size",
+                      "secure_channel",
+                      "client_cert",
+                      "client_key",
+                      "ca_cert",
+                      "login_type"}
 
     def __init__(self):
-        #
-        # Get config from the env variables or from jupyter_cs3_config.json file
-        #
-        if 'CS3_EXT_CONFIG' in os.environ.keys() and utils.strtobool(os.environ['CS3_EXT_CONFIG']):
-            self.config = {}
-            self.log.info("Reading CS3 config from environment")
-            for name in self.__env_names:
-                env_name = "CS3_" + name.upper()
-                if env_name in os.environ:
-                    self.log.info(f"Reading config value {name} = " + str(os.environ[env_name]))
-                    self.config[name] = os.environ[env_name]
-                else:
-                    self.log.error(f"Unable to read {name}")
-                    raise ValueError(f"Unable to read {name} from environment variables")
+        if 'CS3_ENV_CONF' in os.environ.keys() and utils.strtobool(os.environ['CS3_ENV_CONF']):
```

Comment:
We shouldn't use `CS3_ENV_CONF`. This is more os less the same as having `CS3_EXT_CONFIG`.
My proposal is to return to the previous behavior: 
1. we start by loading default values, 
1. if the configuration file exists we load it and replace the values,
1. and if ENV vars exists, we also replace the corresponding values.